src/Uno.UI.Runtime.Skia.X11/X11XamlRootHost.cs

```diff
@@ -167,7 +167,7 @@ private void UpdateWindowPropertiesFromPackage()
 					this.Log().Info($"Loading icon file [{iconPath}] from Package.appxmanifest file");
 				}
 
-				SetIconFromFile(iconPath);
+				Task.Run(() => SetIconFromFile(iconPath));
```

Comment:
Why not do all thing in background thread? Such as check the `iconPath` file exists and so on.

Reply:
Agreed, it's probably also best to extract that part of the current method into a separate one.